### ISSUE TITLE
MiniZinc: internal error: isa: nullptr with MiniSearch

### ISSUE THREAD

**Issue by Alexander-Schiendorfer** [MiniZinc/libminizinc#62]
I'm trying to implement search strategies that rely on set orderings in **MiniSearch** and faced some issues. I produced a minimal working example to show the problem.

The task is to decide a set of int variable x between {} and 1..5 such that only even numbers occur in x. For the sake of illustration, a search strategy enumerates from 1..5 downwards to {}. 

A *working* example for MiniZinc (>= 2.0.2) is found here https://gist.github.com/Alexander-Schiendorfer/4b98df990cf03bd9f4af#file-working-in-minizinc .

However, in MiniSearch, there seems to be a problem with that, shown in the source code https://gist.github.com/Alexander-Schiendorfer/9f95d4ae97f88892f682#file-non-working-example-for-minisearch where the `onlyEvens(x)` constraint is posted after the first solution is found (adapted from BAB). I got the following error

```
Violated scs now: 1..5
MiniZinc: internal error: isa: nullptr
```
I used the latest version obtained from https://github.com/MiniZinc/libminizinc/tree/feature/minisearch .
Since the error is not present in the current MiniZinc version, I hope that it has already been resolved and could maybe applied easily to MiniSearch?

**Comment by Alexander-Schiendorfer** [comment]
FYI: 

Restating

```
predicate onlyEvens(var set of range: y) = 
(
 forall(r in y) (r mod 2 = 0)
);
```
as 
```
predicate onlyEvens(var set of range: y) = 
(
 forall(r in range) (r in y -> r mod 2 = 0)
);
```
also does the trick. Perhaps it has to do with option types?

**Comment by Alexander-Schiendorfer** [comment]
Fixed with merge from master 18a7cb257babb23c1cc24eccd1e6da2c9f54f2b3
